src/controllers/AccessRequestController.ts

```diff
@@ -116,4 +133,254 @@ export class AccessRequestController extends Controller {
+        }
+        // Sort results
+        // results = this.sortDetailsResults(results);
+        return results;
+    }
+
+    @SuccessResponse('204', 'No content')
+    @Put('')
+    /**
+     * Create a new access request for a user
+     */
+    public async updateAccessRequest(
+        @Res() typeORMErrorResponse: TsoaResponse<422, GenericTypeORMErrorType>,
+        @Res()
+        requiredFieldErrorResponse: TsoaResponse<422, requiredFieldErrorType>,
+        @Res() serverErrorResponse: TsoaResponse<500, serverErrorType>,
+        @Res() forbiddenErrorResponse: TsoaResponse<403, forbiddenError>,
+        @Res() notFoundErrorResponse: TsoaResponse<404, notFoundError>,
+        @Body() requestBody: accessRequestUpdateRequestBody,
+        @Request() req: req,
+    ): Promise<void> {
+        this.setStatus(204);
+        let permissions: string[] = [];
+
+        // validate access
+        try {
+            permissions = decodingJWT(req.cookies.token)?.payload.permissions;
+            if (!permissions.includes('ACCESS_REQUEST')) {
+                throw new AuthenticationError(
+                    `Permission 'ACCESS_REQUEST' is not available for this user`,
+                    403,
+                );
+            }
+        } catch (err) {
+            if (err instanceof AuthenticationError) {
+                logger.warn(
+                    `Encountered 403 forbidden error in getAllRequests: ${err.message}`,
+                );
+                return forbiddenErrorResponse(403, {
+                    message: err.message,
+                    code: 403,
+                });
+            }
+            if (err instanceof Error) {
+                logger.warn(
+                    `Encountered 404 not found error in getAllRequests: ${err.message}`,
+                );
+                return notFoundErrorResponse(404, {
+                    message: err.message,
+                    code: 404,
+                });
+            }
+        }
+        // validate inputs
+        if (
+            requestBody.action === requestStatusType.Rejected &&
+            requestBody.rejectionReason === ''
+        ) {
+            const message = 'Must provide reason for rejection.';
+            logger.warn(message);
+            return requiredFieldErrorResponse(422, { message });
+        }
+
+        if (requestBody.action === null || requestBody.action === undefined) {
+            const message = 'Must provide an action for update.';
+            logger.warn(message);
+            return requiredFieldErrorResponse(422, { message });
+        }
+
+        if (requestBody.requestIds.length < 1) {
+            const message = 'Must provide at least of request id';
+            logger.warn(message);
+            return requiredFieldErrorResponse(422, { message });
+        }
+        try {
+            await updateRequestStatus(requestBody);
+            // TODO: add send email functionality
+            /* let emailAddresses: any[] = [];
```

Comment:
Is this supposed to be commented out? I'm assuming this might be because the admin email is not whitelisted?

Reply:
yes. i will push the changes separately